src/pty_bus.rs

```diff
@@ -34,6 +34,26 @@ fn _debug_log_to_file (message: String) {
     file.write_all("\n".as_bytes()).unwrap();
 }
 
+fn debug_to_file(message: u8, pid: RawFd) {
+    use std::fs::OpenOptions;
+    use std::io::prelude::*;
+    let mut path = PathBuf::new();
+    path.push(
+        [
+            String::from("/tmp/mosaic-logs/mosaic-"),
+            pid.to_string(),
+            String::from(".log"),
+        ]
+        .join(""),
```

Comment:
Just a very tiny Rust nitpick, but for joining with an empty string, there is [the `.concat()` method](https://doc.rust-lang.org/std/primitive.slice.html#method.concat) that might be a bit more idiomatic :)